src/app/orderManagement/orders/js/orders.config.js

```diff
@@ -18,10 +18,35 @@ function OrdersConfig($stateProvider) {
                     return ocParameters.Get($stateParams);
                 },
                 OrderList: function(ocOrdersService, OrderCloud, Parameters) {
-                    return ocOrdersService.List(Parameters);
+                    return ocOrdersService.List(Parameters)
+                        .then(function(orders) {
+                            if(orders.Items.length) {
+                                return orders;
+                            } else {
+                                return null;
+                            }
+                        });
                 },
                 BuyerCompanies: function(OrderCloud) {
                     return OrderCloud.Buyers.List(null, 1, 100);
+                },
+                UserGroupsList: function($q, OrderCloud, Parameters, BuyerCompanies) {
+                    var queue = [];
+                    _.each(BuyerCompanies.Items, function(buyer) {
+                        queue.push(function(){
+                            var defer = $q.defer();
```

Comment:
don't use $q.defer();

The reason we don't want to use them is that it wraps something that is already a promise, in another promise. If there are any errors, they aren't correctly propagated to the error handler and cause hard to debug situations.